### DISCUSSION TITLE
Redis SSL Failing Silently

### DISCUSSION THREAD

**Discussion by msmithGH** [miguelgrinberg/Flask-SocketIO#1887] [Q&A]
Hi,

I use Flask-SocketIO in AWS inside of ECS/Fargate.  I establish my connection with:

socketio = SocketIO(app, cors_allowed_origins=[], async_handlers=True, message_queue=redis_address, logger=True, engineio_logger=True)
socketio.init_app(app)

This works, however if I enable "Encryption in transit" on my redis instance (managed by AWS, no additional AUTH password was used in this instance), it fails silently.  Any message emitted to the client after the connection has been established is never received by the client.  There aren't any errors on the client or the server that I can find.  "Encryption at rest" does work fine.  I've tried using the redis address with both rediss:// and redis:// but have the same results.

Are there any configuration steps I need to set in order to tell Flask-SocketIO that my redis message queue uses SSL?

Thanks

**Comment by miguelgrinberg** [comment]
I honestly don't know. If the encryption in transit option that you enabled just means that the Redis server uses SSL, then you should be able to connect to it using a `rediss://` URL instead of `redis://`. If that does not work, then there must be some other option that AWS requires. The URL that you provide for your Redis store is passed as is to the `redis-py` library, so you need to find out if any other options need to be passed in the URL. The SSL documentation is located [here](https://redis-py.readthedocs.io/en/stable/examples/ssl_connection_examples.html). The `RedisManager` class also accepts additional options that are passed along with the URL to the redis library, documentation is [here](https://python-socketio.readthedocs.io/en/latest/api.html#redismanager-class).
